### PR TITLE
Prepare bugfix release

### PR DESCRIPTION
**Pull request by kbattocchi**
This ought to fix reliance on newer sklearn versions, and also cleans up a few documentation issues.

### CODE REVIEW

File: econml/sklearn_extensions/ensemble.py [hunk 1]
```diff
@@ -590,10 +586,9 @@ def _mean_fn(self, X, fn, acc, slice=None):
             n_estimators = slice[1] - slice[0]
 
         # Assign chunk of trees to jobs
-        n_jobs, _, _ = _partition_estimators(n_estimators, self.n_jobs)
+        n_jobs = min(effective_n_jobs(self.n_jobs), n_estimators)
         lock = threading.Lock()
-        Parallel(n_jobs=n_jobs, verbose=self.verbose,
-                 **_joblib_parallel_args(require="sharedmem"))(
+        Parallel(n_jobs=n_jobs, verbose=self.verbose, require="sharedmem")(
```
Comment (vsyrgkanis) [inline econml/sklearn_extensions/ensemble.py:591]: I believe joblib args is doing sth more than just setting require=sharedmem. I believe it also sets backend='threading' when require='sharedmem' is set. Can we verify that the same functionality is eanbled.

File: econml/sklearn_extensions/ensemble.py [hunk 2]
```diff
@@ -555,8 +552,7 @@ def fit(self, X, y, sample_weight=None, sample_var=None):
                                                                        int(np.ceil(self.subsample_fr_ *
                                                                                    (X.shape[0] // 2))),
                                                                        replace=False)])
-            res = Parallel(n_jobs=self.n_jobs, verbose=self.verbose,
-                           **_joblib_parallel_args(prefer='threads'))(
+            res = Parallel(n_jobs=self.n_jobs, verbose=self.verbose, prefer='threads')(
```
Comment (vsyrgkanis) [inline econml/sklearn_extensions/ensemble.py:555]: Similarly here, can we verify that _joblib_parallel_args only returns prefer='threads' and not some other keyword too.

File: econml/sklearn_extensions/ensemble.py [hunk 3]
```diff
@@ -462,7 +459,7 @@ def fit(self, X, y, sample_weight=None, sample_var=None):
         """
 
         # Validate or convert input data
-        X = check_array(X, accept_sparse="csc", dtype=DTYPE)
+        X = check_array(X, accept_sparse="csc", dtype=np.float32)
```
Comment (vsyrgkanis) [inline econml/sklearn_extensions/ensemble.py:462]: We are using the _tree backend of sklearn. Would it be a problem if they change the native dtype in their _tree implementation. I suspect it's benign and we'll discover it.

**Review by vsyrgkanis — APPROVED** [review]
Looks good. Two minor comments

File: econml/sklearn_extensions/ensemble.py (hunk 1 above)
Comment (vsyrgkanis) [reply econml/sklearn_extensions/ensemble.py:591]: this is handled in the joblib versions we require

File: econml/sklearn_extensions/ensemble.py (hunk 2 above)
Comment (vsyrgkanis) [reply econml/sklearn_extensions/ensemble.py:555]: this is handled in the joblib versions we require

File: econml/sklearn_extensions/ensemble.py (hunk 1 above)
Comment (kbattocchi) [reply econml/sklearn_extensions/ensemble.py:591]: There is some extra logic in the sklearn private method, but it's only applicable to versions of joblib<=0.11 while we require joblib 0.13 (which does not need their fix)

File: econml/sklearn_extensions/ensemble.py (hunk 3 above)
Comment (kbattocchi) [reply econml/sklearn_extensions/ensemble.py:462]: Yes, to some extent we have just moved the problem around somewhat; however as you say it seems like we would discover the problem if they change datatypes and it's one less internal thing that we're referencing this way.

**Pull request merged**